core/services/fluxmonitor/flux_monitor.go

```diff
@@ -403,10 +411,18 @@ func (p *PollingDeviationChecker) Start(ctx context.Context, client eth.Client)
 	}
 
 	ctx, p.cancel = context.WithCancel(ctx)
-	go p.consume(ctx, roundSubscription, client)
+	go p.consume(ctx, roundSubscription)
 	return nil
 }
 
+// Stop stops this instance from polling, cleaning up resources.
+func (p *PollingDeviationChecker) Stop() {
```

Comment:
Is it concurrent-safe?

Reply:
Should be, yeah, but it's also a programming error to call Start and Stop more than once.